setup/install/providers/aws/aws-ec2.md

```diff
@@ -102,47 +113,252 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
+4. Edit the trust relationship. Use the following policy, making sure to replace the example ARN with the actual ARN from [Create AWS IAM User](#create-aws-iam-user):
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": [
+             "arn:aws:iam::100000000001:user/spinnakerManaging"
+           ]
+         },
+         "Action": "sts:AssumeRole"
+       }
+     ]
+   }
+   ```
+5. Repeat this process for any managed accounts. This example requires the both **develop** and **staging** to have a trust relationship with the IAM user **spinnakerManaging**.
+
+After you set up the AWS IAM user, roles, policies and trust relationship, enable the AWS Provider in the Spinnaker. You can use the CLI tool *Halyard* to configure and enable the AWS Provider.
+
 ## Halyard Configurations
-After the AWS IAM user, roles, policies and trust relationship have been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
 
-1. Access the Halyard Pod.
-2. Add the configurations for AWS provider with `hal` command. Please check [hal config provider AWS](https://www.spinnaker.io/reference/halyard/commands/#hal-config-provider-aws).
-3. Enable the AWS provider `hal config provider aws enable`.
+After the AWS IAM structure (user, roles, policies, and trust relationship) has been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
 
-### Configure Halyard to use AccessKeys (if configured)
+### Enable the AWS provider
 
-> These steps need to be carried out only if you selected UseAccessKeyForAuthentication as true in Option-1 or Option-2 above
+Before you start, you must have Halyard installed. Additionally, you should understand how to run Halyard commands. Specifically, how you access Halyard depends on how and where you installed Halyard. For example, if you installed Halyard in a Docker container, you need to use the `docker exec` command.
 
-```bash
-hal config provider aws edit --access-key-id ${ACCESS_KEY_ID} \
-    --secret-access-key # do not supply the key here, you will be prompted
-```
+1. Add the AWS accounts to Spinnaker:
 
-### Configure Halyard to add AWS Accounts
+   ```bash
+   hal config provider aws [parameters] [subcommands]
+   ```
 
-```bash
-$AWS_ACCOUNT_NAME={name for AWS account in Spinnaker, e.g. my-aws-account}
+   For information about the available parameters, see [hal config provider AWS](https://www.spinnaker.io/reference/halyard/commands/#hal-config-provider-aws).
 
-hal config provider aws account add $AWS_ACCOUNT_NAME \
-    --account-id ${ACCOUNT_ID} \
-    --assume-role role/spinnakerManaged
-```
+   The following examples add the ****spinnakerManaged** and **develop** accounts from the previous examples. Repeat the command for every account you want to add.
+
+   ```bash
+   # Adds spinnakerManaged for the regions us-east-1 and us-west-2
+   export AWS_ACCOUNT_NAME=""
```

Comment:
Two comments:
* I think this is the correct syntax for setting a variable `export <variable-name> = "<some-name>". You had `$<variable-name>`. Let me know if that method actually works. I'm used to the `export` method.
* Is it intentional that the AWS_ACCOUNT_NAME is blank for spinnakerManaging (arn 100000000001)?

Reply:
I have fixed those issues and added the example ARN